Sources/Sentry/SentryBaseIntegration.m

```diff
@@ -1,14 +1,32 @@
 #import "SentryBaseIntegration.h"
+#import "SentryClient+Private.h"
 #import "SentryCrashWrapper.h"
+#import "SentryHub+Private.h"
 #import "SentryLog.h"
+#import "SentrySDK+Private.h"
 #import <Foundation/Foundation.h>
 #import <SentryDependencyContainer.h>
 #import <SentryOptions+Private.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@interface
+SentryBaseIntegration ()
+@property (strong, nonatomic) SentryCrashWrapper *crashWrapper;
+@end
+
 @implementation SentryBaseIntegration
 
+- (instancetype)initWithCrashWrapper:(SentryCrashWrapper *)crashWrapper
+{
+    if (self = [super init]) {
+        return nil;
+    }
+
+    _crashWrapper = crashWrapper;
+    return self;
+}
```

Comment:
h: I don't like this, most integration don't need crashWrapper (If Im not mistaken only one need it), and now all of them will have this constructor, and a extra crashWrapper property for no reason.

Reply:
There was actually only one place it was even used in this class, and i didn't even use this property for it, so i didn't even need to make the change for the constructor 🤦🏻

Reply:
Honestly, the entire function SentryBaseIntegration.shouldBeEnabledWithOptions should just be moved to SentryClient instead anyways. Then we can remove the param from the init and this is a nonissue.